src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1277,37 +1277,36 @@ UpdateColocationGroupReplicationFactor(uint32 colocationId, int replicationFacto
 	/* we first search for colocation group by its colocation id */
 	pgDistColocation = heap_open(DistColocationRelationId(), RowExclusiveLock);
 	tupleDescriptor = RelationGetDescr(pgDistColocation);
-	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_colocationid, BTEqualStrategyNumber,
-				F_OIDEQ, ObjectIdGetDatum(colocationId));
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_distributioncolumntype,
```

Comment:
I think it might make sense to have an intermediate variable for `InvalidOid` just for readability purposes and create the opportunity to explain the hack we're doing:

```C


/* 
 * Reference tables don't have a distribution column, the distribution column type is invalid. 
  * All reference tables should have the same colocation group with distribution column type is invalid.
  *

  */
referenceTableDistributionColumnType = InvalidOid;

...

F_OIDEQ, ObjectIdGetDatum(
referenceTableDistributionColumnType));

```

If you don't prefer to do it in the function itself, maybe add such a comment to the function comment part. My goal here is to explicitly mention that there is a single entry in the pg_dist_colocation for reference tables, and you can get that via ` distributioncolumntype = 0;`